hail/python/hail/linalg/blockmatrix.py

```diff
@@ -1853,6 +1857,160 @@ def export_rectangles(self, path_out, rectangles, delimiter='\t', binary=False):
         Env.backend().execute(BlockMatrixWrite(self._bmir, writer))
 
+    @typecheck_method(path_out=str, delimiter=str, binary=bool)
+    def export_blocks(self, path_out, delimiter='\t', binary=False):
+        """Export each block of the block matrix as its own delimited text or binary file.
+        This is a special case of :meth:`.export_rectangles`
+
+        Examples
+        --------
+        Consider the following block matrix:
+
+        >>> import numpy as np
+        >>> nd = np.array([[ 1.0, 2.0, 3.0],
+        ...                [ 4.0, 5.0, 6.0],
+        ...                [ 7.0, 8.0, 9.0]])
+
+        >>> BlockMatrix.from_numpy(nd, block_size=2).export_blocks('output/example')
+
+        This produces four files in the folder ``output/example``.
+
+        The first file is ``rect-0_0-2-0-2``:
+
+        .. code-block:: text
+
+            1.0 2.0
+            4.0 5.0
+
+        The second file is ``rect-1_0-2-2-3``:
+
+        .. code-block:: text
+
+            3.0
+            6.0
+
+        The third file is ``rect-2_2-3-0-2``:
+
+        .. code-block:: text
+
+            7.0 8.0
+
+        And the fourth file is ``rect-3_3-4-3-4``:
+
+        .. code-block:: text
+
+            9.0
+
+        Notes
+        -----
+        This method is particularly useful if the block matrix is too large to write to a single file.
+
+        See Also
+        --------
+        :meth:`.rectangles_to_numpy`
+
+        Parameters
+        ----------
+        path_out: :obj:`str`
+            Path for folder of exported files.
+        delimiter: :obj:`str`
+            Column delimiter.
+        binary: :obj:`bool`
+            If true, export elements as raw bytes in row major order.
+        """
+        def rows_in_block(block_row):
+            if block_row == n_block_rows - 1:
+                return self.n_rows - block_row * self.block_size
+            return self.block_size
+
+        def cols_in_block(block_col):
+            if block_col == n_block_cols - 1:
+                return self.n_cols - block_col * self.block_size
+            return self.block_size
+
+        def bounds(block_row, block_col):
+            start_row = block_row * self.block_size
+            start_col = block_col * self.block_size
+            end_row = start_row + rows_in_block(block_row)
+            end_col = start_col + cols_in_block(block_col)
+
+            return [start_row, end_row, start_col, end_col]
+
+        n_block_rows = math.ceil(self.n_rows / self.block_size)
+        n_block_cols = math.ceil(self.n_cols / self.block_size)
+        block_indices = itertools.product(range(n_block_rows), range(n_block_cols))
+        rectangles = [bounds(block_row, block_col) for (block_row, block_col) in block_indices]
+
+        self.export_rectangles(path_out, rectangles, delimiter, binary)
+
+    @staticmethod
+    @typecheck(path=str, binary=bool)
+    def rectangles_to_numpy(path, binary=False):
```

Comment:
I think we need a note for the platform-dependentness of this method in binary mode